lib/lightning/extensions/usage_limiting.ex

```diff
@@ -37,15 +38,55 @@ defmodule Lightning.Extensions.UsageLimiting do
     defstruct [:project_id, :user_id]
   end
 
+  @doc """
+  Checks the usage limits for a given project and user context, returning an error
+  message if the limits are exceeded.
+
+  Requires a `Context` struct containing the `project_id` and `user_id`.
+
+  ## Returns
+
+    - `{:ok}` if the limits are within bounds.
+    - `{:error, reason, message}` if the limits are exceeded.
+  """
   @callback check_limits(context :: Context.t()) ::
               :ok | error()
 
+  @doc """
+  Limits specific actions based on an `Action` and `Context`.
+
+  ## Returns
+
+    - `:ok` if the action is allowed.
+    - `{:error, reason, message}` if the action is not allowed.
+  """
   @callback limit_action(action :: Action.t(), context :: Context.t()) ::
               :ok | error()
 
+  @doc """
+  Increments the AI query count for a given chat session.
+
+  Requires a `Lightning.AiAssistant.ChatSession` struct.
+
+  ## Returns
+
+    - An Ecto.Multi struct representing the transaction to increment AI queries.
+  """
   @callback increment_ai_queries(Lightning.AiAssistant.ChatSession.t()) ::
               Ecto.Multi.t()
 
+  @doc """
+  Returns run options based on the given project context. The run options include
+  a timeout value and a flag for saving dataclips.
+
+  Requires a `Context` struct containing the `project_id`.
+
+  ## Returns
+
+    - A keyword list of run options including:
+      - `:run_timeout_ms`: The run timeout in milliseconds.
+      - `:save_dataclips`: A boolean indicating whether to save dataclips.
+  """
```

Comment:
FYI @elias-ba, I moved the docs you had to the behaviour, that way _all_ the implementations get the same docs.